src/equations/svaerd_kalisch_1d.jl

```diff
@@ -211,18 +283,54 @@ function create_cache(mesh, equations::SvaerdKalischEquations1D,
     return cache
 end
 
-function assemble_system_matrix!(cache, h, D1, D1mat,
-                                 ::SvaerdKalischEquations1D)
-    (; M_h, minus_MD1betaD1) = cache
-
-    @.. M_h = h
-    scale_by_mass_matrix!(M_h, D1)
-
-    # Floating point errors accumulate a bit and the system matrix
-    # is not necessarily perfectly symmetric but only up to
-    # round-off errors. We wrap it here to avoid issues with the
-    # factorization.
-    return Symmetric(Diagonal(M_h) + minus_MD1betaD1)
+# Reflecting boundary conditions assume alpha = gamma = 0
+function create_cache(mesh, equations::SvaerdKalischEquations1D,
+                      solver, initial_condition,
+                      ::BoundaryConditionReflecting,
+                      RealT, uEltype)
+    if !isapprox(equations.alpha, 0, atol = 1e-14) ||
+       !isapprox(equations.gamma, 0, atol = 1e-14)
+        throw(ArgumentError("Reflecting boundary conditions for Svärd-Kalisch equations only implemented for alpha = gamma = 0"))
+    end
```

Comment:
Wouldn't it be simpler to test

```suggestion
    if !iszero(equations.alpha) || !iszero(equations.gamma,)
        throw(ArgumentError("Reflecting boundary conditions for Svärd-Kalisch equations only implemented for alpha = gamma = 0"))
    end
```

The current test disallows `eps(Float32)` deviations for 32-bit computations anyway.